src/librustdoc/clean/types.rs

```diff
@@ -786,7 +786,11 @@ impl Item {
                             // because it isn't public API.
                             None
                         }
-                        _ => Some(rustc_hir_pretty::attribute_to_string(&tcx, attr)),
+                        _ => Some({
+                            let mut s = rustc_hir_pretty::attribute_to_string(&tcx, attr);
+                            assert_eq!(s.pop(), Some('\n'));
```

Comment:
Why not using `trim_end` instead?

Reply:
That returns a `&str` that borrows the original, so it'd need another allocation to go back to a `String`. (Which probably isn't enough to be meaningful perf-wise, but would make me sad)